modules/geo-layers/src/tile-layer/utils.ts

```diff
@@ -57,24 +58,32 @@ function transformBox(bbox: Bounds, modelMatrix: Matrix4): Bounds {
   return transformedBox;
 }
 
-export function getURLFromTemplate(
-  template: string | string[],
-  properties: TileIndex
-): string | null {
+function stringHash(s) {
+  return s.split('').reduce((a, b) => ((a << 5) - a + b.charCodeAt(0)) | 0, 0);
+}
```

Comment:
I'm not familiar with deck.gl's TS style rules. When do we use literal input and/or output types?

Just noting that in the case of this function, I think `stringHash(s: string): number` would've made the intent of the function easier to parse at a glance.

Reply:
Good catch